app/controllers/owners/restaurants_controller.rb

```diff
@@ -0,0 +1,12 @@
+class Owners::RestaurantsController < ApplicationController
+
+  def index
+    @owner = current_user.id
```

Comment:
After going through all your code, you're actually not using `@owner_id` anywhere so get rid of it and the before filter can now become

```ruby
before_action :owner, only: :index
```